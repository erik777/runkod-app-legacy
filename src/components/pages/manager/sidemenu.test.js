import React from 'react';
import TestRenderer from 'react-test-renderer';

import SideMenu from './sidemenu';

test('1- 2 projects, no project selected, not collapsed', () => {
  const props = {
    projects: {
      list: [
        {_id: 'uniqueid1', name: 'project 1'},
        {_id: 'uniqueid2', name: 'project 2'}
      ]
    },
    project: null,
    ui: {
      sideProjectsVisible: true
    },
    selectProject: () => {
    },
    fetchFiles: () => {
    },
    toggleUiProp: () => {
    }
  };

  const renderer = TestRenderer.create(
    <SideMenu {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});

test('2- 2 projects, second project selected, not collapsed', () => {
  const props = {
    projects: {
      list: [
        {_id: 'uniqueid1', name: 'project 1'},
        {_id: 'uniqueid2', name: 'project 2'}
      ]
    },
    project: {_id: 'uniqueid2'},
    ui: {
      sideProjectsVisible: true
    },
    selectProject: () => {
    },
    fetchFiles: () => {
    },
    toggleUiProp: () => {
    }
  };

  const renderer = TestRenderer.create(
    <SideMenu {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});

test('3- 2 projects, collapsed', () => {
  const props = {
    projects: {
      list: [
        {_id: 'uniqueid1', name: 'project 1'},
        {_id: 'uniqueid2', name: 'project 2'}
      ]
    },
    project: {_id: 'uniqueid1'},
    ui: {
      sideProjectsVisible: false
    },
    selectProject: () => {
    },
    fetchFiles: () => {
    },
    toggleUiProp: () => {
    }
  };

  const renderer = TestRenderer.create(
    <SideMenu {...props}/>
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});

