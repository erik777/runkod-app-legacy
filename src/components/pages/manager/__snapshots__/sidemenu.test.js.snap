// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- 2 projects, no project selected, not collapsed 1`] = `
<div
  className="side-menu"
>
  <div
    className="menu-toolbar"
  >
    <a
      className="btn-new-project"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
       
      New Project
    </a>
  </div>
  <div
    className="menu-content"
  >
    <div
      className="menu-list"
    >
      <div
        className="menu-list-header"
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
         
        Projects
      </div>
      <div
        className="menu-items"
      >
        <div
          className="menu-item"
          onClick={[Function]}
        >
          project 1
        </div>
        <div
          className="menu-item"
          onClick={[Function]}
        >
          project 2
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`2- 2 projects, second project selected, not collapsed 1`] = `
<div
  className="side-menu"
>
  <div
    className="menu-toolbar"
  >
    <a
      className="btn-new-project"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
       
      New Project
    </a>
  </div>
  <div
    className="menu-content"
  >
    <div
      className="menu-list"
    >
      <div
        className="menu-list-header"
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
         
        Projects
      </div>
      <div
        className="menu-items"
      >
        <div
          className="menu-item"
          onClick={[Function]}
        >
          project 1
        </div>
        <div
          className="menu-item active"
          onClick={[Function]}
        >
          project 2
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`3- 2 projects, collapsed 1`] = `
<div
  className="side-menu"
>
  <div
    className="menu-toolbar"
  >
    <a
      className="btn-new-project"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
       
      New Project
    </a>
  </div>
  <div
    className="menu-content"
  >
    <div
      className="menu-list"
    >
      <div
        className="menu-list-header"
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
         
        Projects
      </div>
    </div>
  </div>
</div>
`;

exports[`4 redirected projects 1`] = `
<div
  className="side-menu"
>
  <div
    className="menu-toolbar"
  >
    <a
      className="btn-new-project"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
       
      New Project
    </a>
  </div>
  <div
    className="menu-content"
  >
    <div
      className="menu-list"
    >
      <div
        className="menu-list-header"
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
         
        Projects
      </div>
      <div
        className="menu-items"
      >
        <div
          className="menu-item"
          onClick={[Function]}
        >
          project 1
        </div>
        <div
          className="menu-item redirected active"
          onClick={[Function]}
        >
          project 3
        </div>
        <div
          className="menu-item redirected"
          onClick={[Function]}
        >
          project 4
        </div>
        <div
          className="menu-item"
          onClick={[Function]}
        >
          project 2
        </div>
        <div
          className="menu-item redirected"
          onClick={[Function]}
        >
          project 5
        </div>
        <div
          className="menu-item"
          onClick={[Function]}
        >
          project 6
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`5 redirected projects with broken link 1`] = `
<div
  className="side-menu"
>
  <div
    className="menu-toolbar"
  >
    <a
      className="btn-new-project"
      onClick={[Function]}
    >
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
       
      New Project
    </a>
  </div>
  <div
    className="menu-content"
  >
    <div
      className="menu-list"
    >
      <div
        className="menu-list-header"
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
         
        Projects
      </div>
      <div
        className="menu-items"
      >
        <div
          className="menu-item active"
          onClick={[Function]}
        >
          project 1
        </div>
        <div
          className="menu-item"
          onClick={[Function]}
        >
          project 2
        </div>
      </div>
    </div>
  </div>
</div>
`;
